apis/apiextensions/v1/composition_common.go

```diff
@@ -286,15 +286,15 @@ type ContainerFunction struct {
 	// +kubebuilder:validation:Enum="IfNotPresent";"Always";"Never"
 	ImagePullPolicy *corev1.PullPolicy `json:"imagePullPolicy,omitempty"`
 
+	// ImagePullSecrets are used to pull images from private OCI registries.
+	// +optional
+	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
```

Comment:
do we want to specify here patchStrategy and patchMergeKey, similar to what is defined in `PodSpec`?

Reply:
Can you provide more context about what this does and why we'd do it?

I believe you're asking whether we should add these as comment markers / struct tags, and I think the answer is no. My understanding is that these comment markers control strategic merge patching, which isn't supported by CRDs. See:
* https://github.com/kubernetes-sigs/controller-runtime/blob/116a1b8/pkg/client/patch.go#LL172C4-L172C123
* https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#advanced-features-and-flexibility

Strategic merge patches are deprecated in favor of [server-side-apply](https://kubernetes.io/docs/reference/using-api/server-side-apply/), which does support CRDs but isn't something we support yet in Crossplane per https://github.com/crossplane/crossplane/issues/4047.